src/main/java/core/basesyntax/service/PurchaseService.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service;
+
+import core.basesyntax.FruitTransaction;
+import core.basesyntax.db.DataBase;
+
+public class PurchaseService implements OperationHandler {
+    @Override
+    public void processTransaction(FruitTransaction fruitTransaction) {
+        if (DataBase.FRUIT_DATABASE.get(fruitTransaction.getFruit())
+                >= fruitTransaction.getQuantity()) {
+            DataBase.FRUIT_DATABASE.merge(fruitTransaction.getFruit(),
+                    fruitTransaction.getQuantity(),
+                    (existingValue, newValue) -> existingValue - newValue);
+        } else {
+            throw new RuntimeException("Sorry, but we haven't enough "
+                    + fruitTransaction.getFruit());
+        }
```

Comment:
Returning early is a preferred approach in SWE. In practice it means that in such cases you should invert your condition, throw exception, and unwrap the rest of your code, we don't need else after returning/throwing